packages/playwright-test/src/ipc.ts

```diff
@@ -77,6 +77,19 @@ export type StepBeginPayload = {
   forceNoParent: boolean;
   wallTime: number;  // milliseconds since unix epoch
   location?: { file: string, line: number, column: number };
+  serializedTestResult: SerializedTestResult;
+};
+
+export type SerializedTestResult  = Omit<TestResult, 'attachments' | 'startTime' | 'stderr' | 'stdout' | 'steps' > & {
```

Comment:
It seems like you have these new types and utilities that were not necessary. I'd expect that the only thing this PR does is makes sure that TestResult has up-to-date collection of attachments when onStepBegin/End are invoked.